src/cargo/util/lints.rs

```diff
@@ -677,6 +721,41 @@ fn output_unknown_lints(
     Ok(())
 }
 
+/// Note: This only runs on edition 2024+
+///
```

Comment:
I feel this is inaccurate.  This form of unused optional depedencies only exist in 2024+ which is different

Reply:
I don't think this is inaccurate, as [this lint doesn't run if it is less than edition 2024](https://github.com/rust-lang/cargo/blob/a72ce7d278f35e78a168ba5f270974c9f4f91255/src/cargo/util/lints.rs#L698-L701).